src/negotiation.rs

```diff
@@ -583,6 +583,39 @@ where
     }
 }
 
+#[cfg(feature = "serde")]
+impl<Ctx> Serialize for PublicOffer<Ctx>
+where
+    Ctx: Swap,
+{
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s: String = PUB_OFFER_PREFIX.to_owned();
+        let encoded = base58_monero::encode_check(consensus::serialize(self).as_ref())
+            .expect("Encoding in base58 check works");
+        s.push_str(&encoded);
+        serializer.serialize_str(s.as_ref())
```

Comment:
This code will also run if we use `to_string`, we should use it to maintain only one interface IMO

Reply:
agreed - updated